### DISCUSSION TITLE
Save the current session before switching to a new session

### DISCUSSION THREAD

**Discussion by dhruvmanila** [olimorris/persisted.nvim#74] [Ideas]
When switching between sessions using telescope, it should save the current session first and then switch to the new session.

**Comment by olimorris** [comment]
That should be possible already, via [events](https://github.com/olimorris/persisted.nvim#events--callbacks) and notably the `PersistedTelescopeLoadPre` event

**Reply by dhruvmanila** [reply]
```lua
  vim.api.nvim_create_autocmd('User', {
    pattern = 'PersistedTelescopeLoadPre',
    group = group,
    callback = function()
      require('persisted').save()
    end,
    desc = 'persisted.nvim: save current session before switching to another',
  })
```

This is the code I'm using but it's giving me unexpected behavior. I'm currently in session 1 and I try to switch to session 2. Using this, it actually saves the session 2 data in session 1 file thus making both sessions the same.

**Reply by olimorris** [reply]
That's odd as it works perfectly for me with:

```lua
  {
    name = "PersistedHooks",
    {
      "User",
      function(session)
        require("persisted").save()

        -- Delete all of the open buffers
        vim.api.nvim_input("<ESC>:%bd!<CR>")

        -- Don't start saving the session yet
        require("persisted").stop()
      end,
      opts = { pattern = "PersistedTelescopeLoadPre" },
    },
  },
```
